### PR TITLE
Add querys custom

### PR DESCRIPTION
**Pull request by guilleSequeiraWolox**
## Summary
Add querys custom in repositories

## Screenshots
N/A

## JIRA Card
https://trello.com/c/nMrZ7W92/30-queries-custom-en-repositorio

### CODE REVIEW

File: training/src/main/java/wolox/training/repositories/UserRepository.java [hunk 1]
```diff
@@ -12,9 +14,10 @@
 
     Optional<Users> findByUsername(String username);
 
+    @Query(value = "SELECT u FROM Users u WHERE UPPER(name) LIKE UPPER(CONCAT(:name,'%')) and (birthdate BETWEEN :birthdateFrom AND :birthdateTo) ")
```
Comment (rabinovichm) [inline training/src/main/java/wolox/training/repositories/UserRepository.java:18]: It is not wrong, but I think it's not necessary to use concat for `:name` and `%`. Putting `:name%` it's ok.
Also, it should be `%:name%` in order to accept any name containing the provided string,

**Review by rabinovichm — CHANGES_REQUESTED** [review]
Nice job! Just one tiny fix.

File: training/src/main/java/wolox/training/repositories/UserRepository.java (hunk 1 above)
Comment (guilleSequeiraWolox) [reply training/src/main/java/wolox/training/repositories/UserRepository.java:18]: I try to put that. but the next error throw by intellij.

image


Caused by: java.lang.IllegalArgumentException: Validation failed for query for method public abstract java.util.Optional wolox.training.repositories.UserRepository.findByBirthdateBetweenAndNameContainingIgnoreCase(java.time.LocalDate,java.time.LocalDate,java.lang.String)!

Comment (guilleSequeiraWolox) [reply training/src/main/java/wolox/training/repositories/UserRepository.java:18]: I tried different ways but that's the only one that worked for me

Comment (guilleSequeiraWolox) [reply training/src/main/java/wolox/training/repositories/UserRepository.java:18]: another way that worked for me is this:
image

Comment (rabinovichm) [reply training/src/main/java/wolox/training/repositories/UserRepository.java:18]: Oh, sorry! It worked for me, maybe it's the `UPPER`, nevermind.

**Review by rabinovichm — APPROVED** [review]

**Pull request merged**